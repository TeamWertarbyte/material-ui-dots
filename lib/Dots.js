'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _Paper = require('material-ui/Paper');

var _Paper2 = _interopRequireDefault(_Paper);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var styles = {
  dots: {
    position: 'relative'
  },
  dotOuter: {
    float: 'left',
    position: 'absolute'
  },
  dot: {
    transition: 'all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)',
    borderRadius: 4
  }
};

var Dots = function (_Component) {
  _inherits(Dots, _Component);

  function Dots(props) {
    _classCallCheck(this, Dots);

    var _this = _possibleConstructorReturn(this, (Dots.__proto__ || Object.getPrototypeOf(Dots)).call(this, props));

    _this.handleDotClick = function (index, event) {
      if (_this.props.onDotClick != null) {
        _this.props.onDotClick(index, event);
      }
    };

    _this.state = {
      previousIndex: props.index || 0
    };
    return _this;
  }

  _createClass(Dots, [{
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(_ref) {
      var _this2 = this;

      var index = _ref.index,
          dotColor = _ref.dotColor,
          dotSize = _ref.dotSize;

      if (index !== this.props.index) {
        this.setState({ previousIndex: this.props.index });
        this.timeout = setTimeout(function () {
          _this2.timeout = null;
          _this2.setState({ previousIndex: index });
        }, 450);
      }
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      if (this.timeout != null) {
        clearTimeout(this.timeout);
      }
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      var _props = this.props,
          count = _props.count,
          index = _props.index,
          _props$style = _props.style,
          style = _props$style === undefined ? {} : _props$style,
          onDotClick = _props.onDotClick,
          dotColor = _props.dotColor,
          dotSize = _props.dotSize,
          other = _objectWithoutProperties(_props, ['count', 'index', 'style', 'onDotClick', 'dotColor', 'dotSize']);

      var previousIndex = this.state.previousIndex;

      styles.dot = _extends({}, styles.dot, {
        width: dotSize,
        height: dotSize
      });
      styles.dotOuter = _extends({}, styles.dotOuter, {
        width: dotSize,
        height: dotSize,
        padding: dotSize / 2
      });
      return _react2.default.createElement(
        'div',
        _extends({ style: _extends({}, style, { width: count * (dotSize * 2) }) }, other),
        _react2.default.createElement(
          'div',
          { style: styles.dots },
          [].concat(_toConsumableArray(Array(count).keys())).map(function (i) {
            return _react2.default.createElement(
              'div',
              {
                key: i,
                style: _extends({}, styles.dotOuter, {
                  left: i * (dotSize * 2),
                  cursor: onDotClick != null ? 'pointer' : 'inherit'
                }),
                onClick: function onClick(event) {
                  return _this3.handleDotClick(i, event);
                }
              },
              _react2.default.createElement(_Paper2.default, {
                square: false,
                elevation: 0,
                style: _extends({}, styles.dot, {
                  backgroundColor: dotColor,
                  opacity: i >= Math.min(previousIndex, index) && i <= Math.max(previousIndex, index) ? 0 : 0.5
                })
              })
            );
          }),
          _react2.default.createElement(_Paper2.default, {
            elevation: 0,
            style: _extends({}, styles.dot, {
              backgroundColor: dotColor,
              position: 'absolute',
              marginTop: dotSize / 2,
              left: Math.min(previousIndex, index) * (dotSize * 2) + dotSize / 2,
              width: Math.abs(previousIndex - index) * (dotSize * 2) + dotSize
            })
          })
        )
      );
    }
  }]);

  return Dots;
}(_react.Component);

exports.default = Dots;


Dots.propTypes = {
  count: _propTypes2.default.number.isRequired,
  index: _propTypes2.default.number.isRequired,
  style: _propTypes2.default.object,
  dotColor: _propTypes2.default.string,
  dotSize: _propTypes2.default.number,
  onDotClick: _propTypes2.default.func
};

Dots.defaultProps = {
  dotColor: '#fff',
  dotSize: 8
};