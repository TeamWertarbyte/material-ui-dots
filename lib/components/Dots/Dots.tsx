import React, { useEffect, useState } from 'react';
import Paper from '@mui/material/Paper';
import Box from '@mui/material/Box';
import { SxProps, Theme } from '@mui/material/styles';

interface DotsProps {
  count: number;
  index: number;
  sx?: SxProps<Theme>;
  onDotClick?: (
    index: number,
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
  ) => void;
}

export const Dots: React.FC<DotsProps> = ({ count, index, sx, onDotClick }) => {
  const [previousIndex, setPreviousIndex] = useState<number>(index);

  useEffect(() => {
    if (index !== previousIndex) {
      const timeout = setTimeout(() => {
        setPreviousIndex(index);
      }, 400);
      return () => clearTimeout(timeout);
    }
  }, [index, previousIndex]);

  const handleDotClick = (
    dotIndex: number,
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
  ) => {
    if (onDotClick) {
      onDotClick(dotIndex, event);
    }
  };

  return (
    <Box sx={{ width: count * 16, ...sx }}>
      <Box sx={{ position: 'relative', padding: '20px 0 28px' }}>
        {[...Array(count).keys()].map((i) => (
          <Box
            key={i}
            sx={{
              width: 8,
              height: 8,
              padding: 0.5,
              position: 'absolute',
              left: i * 16,
              cursor: onDotClick ? 'pointer' : 'inherit',
            }}
            onClick={(event) => handleDotClick(i, event)}
          >
            <Paper
              elevation={0}
              sx={{
                width: 8,
                height: 8,
                background: '#fff',
                borderRadius: 4,
                transition: 'all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)',
                opacity:
                  i >= Math.min(previousIndex, index) &&
                  i <= Math.max(previousIndex, index)
                    ? 0
                    : 0.5,
              }}
            />
          </Box>
        ))}
        <Paper
          elevation={0}
          sx={{
            position: 'absolute',
            marginTop: 0.5,
            left: Math.min(previousIndex, index) * 16 + 4,
            width: Math.abs(previousIndex - index) * 16 + 8,
            height: 8,
            background: '#fff',
            borderRadius: 4,
            transition: 'all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)',
          }}
        />
      </Box>
    </Box>
  );
};
