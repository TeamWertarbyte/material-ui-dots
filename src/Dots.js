import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Paper from 'material-ui/Paper'

const styles = {
  dots: {
    position: 'relative'
  },
  dotOuter: {
    float: 'left',
    position: 'absolute'
  },
  dot: {
    transition: 'all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)',
    borderRadius: 4
  }
}

export default class Dots extends Component {
  constructor (props) {
    super(props)
    this.state = {
      previousIndex: props.index || 0
    }
  }

  componentWillReceiveProps ({index, dotColor, dotSize}) {
    if (index !== this.props.index) {
      this.setState({previousIndex: this.props.index})
      this.timeout = setTimeout(() => {
        this.timeout = null
        this.setState({previousIndex: index})
      }, 450)
    }
  }

  componentWillUnmount () {
    if (this.timeout != null) {
      clearTimeout(this.timeout)
    }
  }

  handleDotClick = (index, event) => {
    if (this.props.onDotClick != null) {
      this.props.onDotClick(index, event)
    }
  }

  render () {
    const {count, index, style = {}, onDotClick, dotColor, dotSize, ...other} = this.props
    const {previousIndex} = this.state
    styles.dot = {
      ...styles.dot,
      width: dotSize,
      height: dotSize
    }
    styles.dotOuter = {
      ...styles.dotOuter,
      width: dotSize,
      height: dotSize,
      padding: dotSize / 2
    }
    return (
      <div style={{...style, width: count * (dotSize * 2)}} {...other}>
        <div style={styles.dots}>
          {[...Array(count).keys()].map((i) => (
            <div
              key={i}
              style={{
                ...styles.dotOuter,
                left: i * (dotSize * 2),
                cursor: onDotClick != null ? 'pointer' : 'inherit'
              }}
              onClick={(event) => this.handleDotClick(i, event)}
            >
              <Paper
                square={false}
                elevation={0}
                style={{
                  ...styles.dot,
                  backgroundColor: dotColor,
                  opacity: i >= Math.min(previousIndex, index) && i <= Math.max(previousIndex, index) ? 0 : 0.5
                }}
              />
            </div>
          ))}
          <Paper
            elevation={0}
            style={{
              ...styles.dot,
              backgroundColor: dotColor,
              position: 'absolute',
              marginTop: (dotSize / 2),
              left: Math.min(previousIndex, index) * (dotSize * 2) + (dotSize / 2),
              width: Math.abs(previousIndex - index) * (dotSize * 2) + dotSize
            }}
          />
        </div>
      </div>
    )
  }
}

Dots.propTypes = {
  count: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
  style: PropTypes.object,
  dotColor: PropTypes.string,
  dotSize: PropTypes.number,
  onDotClick: PropTypes.func
}

Dots.defaultProps = {
  dotColor: '#fff',
  dotSize: 8
}
