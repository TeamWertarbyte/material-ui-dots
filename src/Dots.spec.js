/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles'
import Dots from './Dots'

const theme = createMuiTheme()

function themed (Component) {
  return (props) => (
    <MuiThemeProvider theme={theme}>
      <Component {...props} />
    </MuiThemeProvider>
  )
}

const ThemedDots = themed(Dots)

test('three dots, first is selected', () => {
  const tree = renderer.create(
    <ThemedDots
      count={3}
      index={0}
    />
  )
  expect(tree).toMatchSnapshot()
})

test('custom style', () => {
  const tree = renderer.create(
    <ThemedDots
      count={5}
      index={0}
      style={{ backgroundColor: 'red' }}
    />
  )
  expect(tree).toMatchSnapshot()
})

test('onDotClick handler', () => {
  const handleDotClick = jest.genMockFunction()
  const tree = renderer.create(
    <ThemedDots
      count={3}
      index={0}
      onDotClick={handleDotClick}
    />
  )
  expect(tree).toMatchSnapshot()
})

test('custom dot color', () => {
  const tree = renderer.create(
    <ThemedDots
      count={4}
      index={0}
      dotColor='red'
    />
  )
  expect(tree).toMatchSnapshot()
})

test('custom dot size', () => {
  const tree = renderer.create(
    <ThemedDots
      count={4}
      index={0}
      dotSize={12}
    />
  )
  expect(tree).toMatchSnapshot()
})
