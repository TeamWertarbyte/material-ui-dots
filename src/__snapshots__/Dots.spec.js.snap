// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom dot color 1`] = `
<div
  style={
    Object {
      "width": 64,
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 0,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 16,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 32,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 48,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "red",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
      style={
        Object {
          "backgroundColor": "red",
          "borderRadius": 4,
          "height": 8,
          "left": 4,
          "marginTop": 4,
          "position": "absolute",
          "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
          "width": 8,
        }
      }
    />
  </div>
</div>
`;

exports[`custom dot size 1`] = `
<div
  style={
    Object {
      "width": 96,
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 12,
          "left": 0,
          "padding": 6,
          "position": "absolute",
          "width": 12,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 12,
            "opacity": 0,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 12,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 12,
          "left": 24,
          "padding": 6,
          "position": "absolute",
          "width": 12,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 12,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 12,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 12,
          "left": 48,
          "padding": 6,
          "position": "absolute",
          "width": 12,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 12,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 12,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 12,
          "left": 72,
          "padding": 6,
          "position": "absolute",
          "width": 12,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 12,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 12,
          }
        }
      />
    </div>
    <div
      className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 4,
          "height": 12,
          "left": 6,
          "marginTop": 6,
          "position": "absolute",
          "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
          "width": 12,
        }
      }
    />
  </div>
</div>
`;

exports[`custom style 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
      "width": 80,
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 0,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 16,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 32,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 48,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 64,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 4,
          "height": 8,
          "left": 4,
          "marginTop": 4,
          "position": "absolute",
          "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
          "width": 8,
        }
      }
    />
  </div>
</div>
`;

exports[`onDotClick handler 1`] = `
<div
  style={
    Object {
      "width": 48,
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "float": "left",
          "height": 8,
          "left": 0,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "float": "left",
          "height": 8,
          "left": 16,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "float": "left",
          "height": 8,
          "left": 32,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 4,
          "height": 8,
          "left": 4,
          "marginTop": 4,
          "position": "absolute",
          "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
          "width": 8,
        }
      }
    />
  </div>
</div>
`;

exports[`three dots, first is selected 1`] = `
<div
  style={
    Object {
      "width": 48,
    }
  }
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 0,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 16,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "inherit",
          "float": "left",
          "height": 8,
          "left": 32,
          "padding": 4,
          "position": "absolute",
          "width": 8,
        }
      }
    >
      <div
        className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 4,
            "height": 8,
            "opacity": 0.5,
            "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
            "width": 8,
          }
        }
      />
    </div>
    <div
      className="MuiPaper-root-1 MuiPaper-elevation0-3 MuiPaper-rounded-2"
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 4,
          "height": 8,
          "left": 4,
          "marginTop": 4,
          "position": "absolute",
          "transition": "all 400ms cubic-bezier(0.4, 0.0, 0.2, 1)",
          "width": 8,
        }
      }
    />
  </div>
</div>
`;
